lib/Driver/Job.cpp

```diff
@@ -106,3 +106,10 @@ void Job::printCommandLine(raw_ostream &os, StringRef Terminator) const {
   printArguments(os, Arguments);
   os << Terminator;
 }
+
+StringRef Job::getCommandLine(StringRef Terminator) const {
+  std::string Out;
+  llvm::raw_string_ostream OS(Out);
+  printCommandLine(OS);
+  return StringRef(Out);
```

Comment:
Please just stick to `printCommandLine`. There's a type `llvm::raw_svector_ostream` which can be used to print to SmallStrings and other SmallVectorImpl<char>.

(Alternately, you could print a summary of the job instead, like "Compile x.swift".)